lib/puppet/provider/mongodb.rb

```diff
@@ -22,8 +23,9 @@ def self.mongo_eval(cmd, db = 'admin')
     if mongorc_file
         cmd = mongorc_file + cmd
     end
-
-    out = mongo([db, '--quiet', '--eval', cmd])
+    config = YAML.load_file('/etc/mongod.conf')
```

Comment:
This module support mongodb 2.4 which uses a key-value style config file, so parsing it as YAML would fail. It also supports different potential file paths described in mongodb::params, so not all installations would be able to find this.

What we could do is check for the mongo version with the mongo_version or mongo_24? methods, check for the presence of either /etc/mongod.conf of /etc/mongodb.conf, and then if we have mongo 2.4 we could get the port with something like this:

```ruby
config = {}
File.readlines('/etc/mongodb.conf').collect do |line|
   k,v = line.split('=')
   config[k.rstrip] = v.lstrip.chomp if k and v
end
port = config['port']
```